### PR TITLE
feat: Add support for navigating in mutator workspaces.

### LINKED ISSUE #9002: Make mutator bubbles focusable and navigable
See parent issue.

### PR DESCRIPTION
**Pull request by gonfunko**
In conjunction with https://github.com/google/blockly/pull/9151, this PR fixes https://github.com/google/blockly/issues/9002 by adding support for navigating into and manipulating blocks within mutator workspaces using keyboard navigation.

### CODE REVIEW

File: test/webdriverio/test/mutator_test.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    await this.browser.pause(PAUSE_TIME);
+    // Hit enter to enter insert mode
+    await this.browser.keys(Key.Enter);
+    await this.browser.pause(PAUSE_TIME);
+    // Hit enter again to lock it into place on the connection
+    await this.browser.keys(Key.Enter);
+
+    const topBlocks = await this.browser.execute(() => {
+      const focusedTree = Blockly.getFocusManager().getFocusedTree();
+      if (!(focusedTree instanceof Blockly.WorkspaceSvg)) return [];
```
Comment (BenHenning) [inline test/webdriverio/test/mutator_test.ts]: Can we make this throw a hard error instead of being a condition? It seems that we probably don't want to default behavior in this case and instead just fail the test.

File: src/navigation.ts [hunk 2]
```diff
@@ -824,9 +824,12 @@ export class Navigation {
    * @returns whether keyboard navigation is currently allowed.
    */
   canCurrentlyNavigate(workspace: Blockly.WorkspaceSvg) {
-    const accessibilityMode = workspace.isFlyout
-      ? workspace.targetWorkspace?.keyboardAccessibilityMode
-      : workspace.keyboardAccessibilityMode;
+    const accessibilityMode = (
+      workspace.getRootWorkspace() ??
+      workspace.targetWorkspace?.getRootWorkspace() ??
```
Comment (BenHenning) [inline src/navigation.ts:836]: These two are new. I'm assuming `targetWorkspace` is only used for flyouts which is why that fallback below is a correct logical replacement to the previous `isFlyout` check, but are these root workspace checks always correct and in this order?

It might help to add a line comment before this explaining when each condition can occur, and why the fallback is correct if more than 2 can occur at once (since `workspace` will always be present).

File: src/actions/exit.ts [hunk 3]
```diff
@@ -39,6 +40,26 @@ export class ExitAction {
               workspace.hideChaff();
             }
             return true;
+          case Constants.STATE.WORKSPACE: {
```
Comment (BenHenning) [inline src/actions/exit.ts:43]: How does escape work when the mutator's flyout is open? At present it seems like it will navigate away from the mutator workspace rather than closing the flyout, or does this work because the navigation state correctly represents flyout/toolbox as needed?

This seems like a good test to add, as well, since it's a specific edge case that could be missed by existing tests if it broke in the future.

File: src/actions/enter.ts [hunk 4]
```diff
@@ -152,6 +153,9 @@ export class EnterAction {
       return true;
     } else if (curNode instanceof icons.Icon) {
       curNode.onClick();
+      renderManagement.finishQueuedRenders().then(() => {
```
Comment (BenHenning) [inline src/actions/enter.ts:159]: I'm slightly surprised this works in all cases, and am wondering if there are cases where it doesn't per this line:

https://github.com/google/blockly/blob/4977b4bc9bd70412456c2d125f5044cd34b625e0/core/icons/mutator_icon.ts#L171

It seems that updating the visibility of the bubble itself blocks on `finishQueuedRenders`, but then queues more? I'm actually unsure of the semantics when calling an async function from a non-async function without using `await`. https://stackoverflow.com/a/53730574 suggests that `finishQueuedRenders` could actually resolve after the rest of the logic occurs, but there's subtlety there with the return value that may not be relevant here.

I'm not sure if there's a better solution, but I think it might be helpful at least to add a comment here explaining why the `finishQueuedRenders` call is both necessary and correct following `onClick` in case we do catch a case in the future where this doesn't work and need to investigate it.

**Review by BenHenning — APPROVED** [review]
Thanks @gonfunko! Had a few thoughts, but the PR largely LGTM. Feel free to re-add me if you'd like another review pass.

File: src/actions/exit.ts (hunk 3 above)
Comment (gonfunko) [reply src/actions/exit.ts:43]: Mutator flyouts aren't closable, but if the mutator flyout is focused the navigation state is indeed `.FLYOUT`, so Escape moves focus to the mutator workspace. A second Escape will then dismiss the mutator bubble.

File: test/webdriverio/test/mutator_test.ts (hunk 1 above)
Comment (gonfunko) [reply test/webdriverio/test/mutator_test.ts]: Done.

File: src/navigation.ts (hunk 2 above)
Comment (gonfunko) [reply src/navigation.ts:836]: That's correct, added a comment explaining the logic here.

File: src/actions/exit.ts (hunk 3 above)
Comment (gonfunko) [reply src/actions/exit.ts:43]: Added the test as well.

File: src/actions/enter.ts (hunk 4 above)
Comment (gonfunko) [reply src/actions/enter.ts:159]: Experimentally it seems to, but I agree this does seem a bit fragile. Added the comment, and if we do run across a situation where this doesn't work we can investigate further at that point.
